java-runtime/ftl-runtime/deployment/src/main/java/xyz/block/ftl/deployment/FtlProcessor.java

```diff
@@ -536,9 +543,33 @@ private static Class<?> loadClass(org.jboss.jandex.Type param) throws ClassNotFo
                 default:
                     throw new RuntimeException("Unknown primitive type " + param.asPrimitiveType().primitive());
             }
-        } else {
-            throw new RuntimeException("Unknown type " + param.kind());
+        } else if (param.kind() == org.jboss.jandex.Type.Kind.ARRAY) {
```

Comment:
nit: `FTLCodeGenerator.java` and this file both feature type projection logic. It might be useful to consolidate this class of logic.

Reply:
One maps from a Jandex bytecode representation of a type to a FTL schema, the other maps the FTL schema representation of a type to a Java language string representing the type. They are not dealing with the same types of objects, and also have subtle differences in the type that is supposed to be generated. e.g. a FTL boolean type should generally just be mapped to a boolean, except if you are producing a type parameter because you can't use primitives in type params.

Reply:
At some point soon I will also add Kotlin to that integration test, so you can see how the calls / schemas map in all 3 languages.

Reply:
The motivation for that comment has to do with the similarity of the type projection task being accomplished. There is probably some desire for this to work consistently across the these type systems and it will be easier to spot unintended drift if they are co-located. I marked this as a nit though because I suspect there are other implementation concerns that might be negatively impacted decoupling this type projection logic